src/librustc_mir/error_codes.rs

```diff
@@ -2144,6 +2144,32 @@ fn main() {
 ```
 "##,
 
+E0594: r##"
+A non-mutable value was assigned a value.
+
+Erroneous code example:
+
+```compile_fail,E0594
+struct SolarSystem {
+    earth: i32,
+}
+
+let ss = SolarSystem { earth: 3 };
+ss.earth = 2; // error!
+```
+
+To fix this error, just declare `ss` as mutable by using the `mut` keyword:
```

Comment:
```suggestion
To fix this error, declare `ss` as mutable by using the `mut` keyword:
```

Reply:
Thanks! I'll update once the error codes have been moved.